runtime/com/redhat/ceylon/compiler/java/Util.java

```diff
@@ -1708,7 +1708,7 @@ private static String notArrayType(Object o) {
     public static int toInt(long value) {
         int result = (int) value;
         if (result != value) {
-            throw new OverflowException();
+            throw new OverflowException(value + " cannot be safely converted into a 32-bit integer");
```

Comment:
Here (and in the other changes in this file) I used "safely converted" instead of the "coerced" proposed by Gavin, as this wording is in the Javadoc up there.

I guess whoever sees this exception doesn't usually actually see this Javadoc, so this reasoning falls a bit short. If wanted, I can change this to "coerced", too.